account_tax_balance/hooks.py

```diff
@@ -0,0 +1,33 @@
+# coding: utf-8
+# © 2020 Opener B.V. <https://opener.amsterdam>
+# © 2020 Graeme Gellatly
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+
+
+def pre_init_hook(cr):
+    """ Precreate move_type and fill with appropriate values to prevent
+    a MemoryError when the ORM attempts to call its compute method on a large
+    amount of preexisting moves. """
+    logger = logging.getLogger(__name__)
+    logger.info('Add account_move.move_type column if it does not yet exist')
+    cr.execute(
+        "ALTER TABLE account_move ADD COLUMN IF NOT EXISTS move_type VARCHAR")
+    cr.execute(
+        """
+        UPDATE account_move am SET move_type = CASE
+            WHEN aa.internal_type = 'liquidity' THEN 'liquidity'
+            WHEN aa.internal_type = 'payable' and aml.balance < 0
+                THEN 'payable'
+            WHEN aa.internal_type = 'payable' and aml.balance >= 0
+                THEN 'payable_refund'
+            WHEN aa.internal_type = 'receivable' and aml.balance > 0
+                THEN 'receivable'
+            WHEN aa.internal_type = 'receivable' and aml.balance <= 0
+                THEN 'receivable_refund'
+            ELSE 'other'
+        END
+        FROM account_move_line aml, account_account aa
+        WHERE aa.id = aml.account_id
+            AND aml.move_id = am.id AND am.move_type IS NULL
+        """)
```

Comment:
@StefanRijnhart I am not sure this is right now I look. Because an account_move will have multiple lines of varying account types, we cannot do this, we need to evaluate the account types in order. With this query the behaviour is indeterminate.

Reply:
You still could do it in one query, but it would require an array_agg grouping on account_type and an in clause.

Reply:
You're totally right, then my diff is better IMO, and with logging.

Reply:
Thanks for noting. Replaced by Pedro's mapping

Reply:
Thanks for noting. Replaced by Pedro's mapping